src/sockref.rs

```diff
@@ -118,16 +94,16 @@ where
     }
 }
 
-/// On Unix, a corresponding `From<&impl AsRawFd>` implementation exists.
+/// On Unix, a corresponding `From<&impl AsFd>` implementation exists.
 #[cfg(windows)]
 #[cfg_attr(docsrs, doc(cfg(windows)))]
 impl<'s, S> From<&'s S> for SockRef<'s>
 where
-    S: AsRawSocket,
+    S: AsSocket,
 {
```

Comment:
You should be careful with this kind of blanket impl. This will prevent you from implementing `From<&T>` for any other type `T` even if `T` does not implement `AsSocket`. For example, the previous impl for `S: AsRawSocket` would prevent you from adding the `S: AsSocket` impl without the breaking change you're currently going through. 

In particular, this prevents you from having `From<&T>` for any type that is `AsRawSocket` but isn't `AsSocket`.

The impl might still be a good idea — as long as you've thought about whether this is acceptable, then that sounds good to me.

Reply:
Two justifications: 

* The `T: AsRawSocket` bound is unsound. Replacing it with `T: AsSocket` makes it sound. It is okay to make breading changes if they are to fix unsoundness
* The bound mirrors the `From<OwnedSocket>` bound on `Socket`.

Reply:
> You should be careful with this kind of blanket impl. This will prevent you from implementing `From<&T>` for any other type `T` even if `T` does not implement `AsSocket`. For example, the previous impl for `S: AsRawSocket` would prevent you from adding the `S: AsSocket` impl without the breaking change you're currently going through.
> 
> In particular, this prevents you from having `From<&T>` for any type that is `AsRawSocket` but isn't `AsSocket`.

I know, that's why I removed the original implementation.
 
> The impl might still be a good idea — as long as you've thought about whether this is acceptable, then that sounds good to me.

The concern above is acceptable because this is the only sound way to create `SockRef`, `AsRaw`/`AsSocket` was exactly designed for this reason. `SockRef` is essentially a type `BorrowedFd` similar to what `TcpStream` is to a `OwnedFd`.